delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala

```diff
@@ -0,0 +1,196 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, IdSegment, Label}
+import ch.epfl.bluebrain.nexus.delta.sdk._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.AclAddress
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resources.ResourceRejection
+import io.circe.Json
+import kamon.instrumentation.akka.http.TracingDirectives.operationName
+import monix.execution.Scheduler
+
+/**
+  * The resource routes
+  *
+  * @param identities the identity module
+  * @param acls       the ACLs module
+  * @param projects   the projects module
+  * @param resources  the resources module
+  */
+final class ResourcesRoutes(
+    identities: Identities,
+    acls: Acls,
+    projects: Projects,
+    resources: Resources
+)(implicit baseUri: BaseUri, s: Scheduler, cr: RemoteContextResolution, ordering: JsonKeyOrdering)
+    extends AuthDirectives(identities, acls)
+    with CirceUnmarshalling {
+
+  import baseUri.prefixSegment
+
+  private val resourceSchema = IriSegment(schemas.resources)
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractCaller { implicit caller =>
+        pathPrefix("resources") {
+          concat(
+            // SSE resources
+            (pathPrefix("events") & pathEndOrSingleSlash) {
+              get {
+                operationName(s"$prefixSegment/resources/events") {
+                  authorizeFor(AclAddress.Root, events.read).apply {
+                    lastEventId { offset =>
+                      emit(resources.events(offset))
+                    }
+                  }
+                }
+              }
+            },
+            projectRef { ref =>
+              routesForProject(ref)
+            },
+            (uuid & uuid) { (orgUuid, projectUuid) =>
+              fetchProject(orgUuid, projectUuid) { project =>
+                routesForProject(project.value.ref)
+              }
+            }
+          )
+        }
+      }
+    }
+
+  private def routesForProject(ref: ProjectRef)(implicit caller: Caller): Route =
+    concat(
+      (pathPrefix("events") & pathEndOrSingleSlash) {
+        get {
+          operationName(s"$prefixSegment/resources/{org}/{project}/events") {
+            authorizeFor(AclAddress.Project(ref), events.read).apply {
+              lastEventId { offset =>
+                emit(resources.events(ref, offset).leftWiden[ResourceRejection])
+              }
+            }
+          }
+        }
+      },
+      // Create a resource without schema nor id segment
+      (post & noParameter("rev") & pathEndOrSingleSlash & entity(as[Json])) { source =>
+        operationName(s"$prefixSegment/resources/{org}/{project}") {
+          authorizeFor(AclAddress.Project(ref), resourcePermissions.write).apply {
+            emit(StatusCodes.Created, resources.create(ref, resourceSchema, source).map(_.void))
+          }
+        }
+      },
+      idSegment { schemaSegment =>
```

Comment:
If we could parse to something like a NoSegment, couldn't we parse the two segments at the same time and make the routes a little simpler ?

Reply:
I think it is perfectly fine as is. More abstraction is just going to make it more complicated. Besides now it is in order:

```scala
idSegment { schemaSegment =>
   // something to do after only the schema segment has been consumed
   idSegment { idSegment =>
      // something to do after both schema and id segment have been consumed
   }
}
```

Reply:
I just find the route a little complicated to understand and am trying to find a way to make it a little simpler. But yes, my solution may not be a good one